src/lib/edf_schedule.c

```diff
@@ -540,6 +540,31 @@ static void schedule_edf_task_free(struct task *task)
 	edf_sch_set_pdata(task, NULL);
 }
 
+/* scheduler of idle tasks (non prioritized) */
+void schedule_idle(void)
+{
+	struct edf_schedule_data *sch =
+		(*arch_schedule_get_data())->edf_sch_data;
+	struct list_item *tlist;
+	struct task *task;
+	struct list_item *clist;
+
+	/* first check if any higher priority task is waiting */
+	schedule();
+
+	/* invoke unprioritized/idle tasks right away */
+	list_for_item_safe(clist, tlist, &sch->idle_list) {
+		task = container_of(clist, struct task, list);
+
+		/* run task if we find any queued */
+		if (task->state == SOF_TASK_STATE_QUEUED)
+			task->func(task->data);
```

Comment:
It's blocking execution on passive level. I think we should invoke `list_item_del(clist)` under this if, because there is no possibility there will be any non executable tasks on the list.